### PR TITLE
Tetrahedron, Icosahedron primitives

### PR DESCRIPTION
**Pull request by kakashihatakae**
Created primitives for Tetrahedon and icosahedron

cube
cube

These are the images of the given shapes.

### CODE REVIEW

File: fury/primitive.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 from fury.transform import cart2sphere, euler_matrix
 from scipy.spatial import ConvexHull
 from scipy.spatial import transform
+import math as m
```
Comment (skoudoro) [inline fury/primitive.py]: Use just `import math`. We should avoid a single letter

File: fury/primitive.py [hunk 2]
```diff
@@ -328,3 +329,85 @@ def _fexp(x, p):
     vertices = np.ascontiguousarray(xyz)
 
     return vertices, sphere_triangles
+
+def prim_tetrahedron():
+
```
Comment (skoudoro) [inline fury/primitive.py]: Can you remove the empty line here

File: fury/primitive.py [hunk 3]
```diff
@@ -328,3 +329,85 @@ def _fexp(x, p):
...
+    '''
```
Comment (skoudoro) [inline fury/primitive.py]: Please, Can you use triple quotes?

File: fury/primitive.py [hunk 4]
```diff
@@ -328,3 +329,85 @@ def _fexp(x, p):
...
+    returns vertices and triangles to a tetrahedron
+    with side two
```
Comment (skoudoro) [inline fury/primitive.py]: `with side two` : What do you mean here?

File: fury/primitive.py [hunk 5]
```diff
@@ -328,3 +329,85 @@ def _fexp(x, p):
...
+    
+
+    Returns
+    -------
+    pyramid_vert: numpy array
```
Comment (skoudoro) [inline fury/primitive.py]: numpy.ndarray

File: fury/primitive.py [hunk 6]
```diff
@@ -328,3 +329,85 @@ def _fexp(x, p):
...
+        4 vertices co-ordinates
+    triangles: ndarray
```
Comment (skoudoro) [inline fury/primitive.py]: same as above

File: fury/primitive.py [hunk 7]
```diff
@@ -328,3 +329,85 @@ def _fexp(x, p):
...
+
+    pyramid_triag = np.array([[1,2,0],
+                          [3,2,0],
+                          [0,3,1],
+                          [1,2,3]], dtype='i8')
+
+    return pyramid_vert, pyramid_triag
+
+def prim_icosahedron():
+    '''
```
Comment (skoudoro) [inline fury/primitive.py]: same comments as above

File: fury/primitive.py [hunk 8]
```diff
@@ -328,3 +329,85 @@ def _fexp(x, p):
...
+    returns vertices and triangles to a icosahedron
+
+    Returns
+    -------
+    icosahedron_vertices: numpy array
```
Comment (skoudoro) [inline fury/primitive.py]: same comments as above

File: fury/primitive.py [hunk 9]
```diff
@@ -328,3 +329,85 @@ def _fexp(x, p):
...
+        12 vertices co-ordinates to the icosahedron
+    icosahedron_mesh: ndarray
```
Comment (skoudoro) [inline fury/primitive.py]: same comments as above

File: fury/primitive.py [hunk 10]
```diff
@@ -328,3 +329,85 @@ def _fexp(x, p):
...
+    return pyramid_vert, pyramid_triag
+
+def prim_icosahedron():
+    '''
+    returns vertices and triangles to a icosahedron
+
+    Returns
+    -------
+    icosahedron_vertices: numpy array
+        12 vertices co-ordinates to the icosahedron
```
Comment (skoudoro) [inline fury/primitive.py]: coordinates instead

File: fury/primitive.py [hunk 11]
```diff
@@ -328,3 +329,85 @@ def _fexp(x, p):
...
+    icosahedron_mesh: ndarray
+        20 triangles representing the tetrahedron
+    '''
+
+
```
Comment (skoudoro) [inline fury/primitive.py]: empty lines to remove

**Comment by kakashihatakae** [comment]
Hello,
Did the needful, will add the test as soon as possible.

**Comment by kakashihatakae** [comment]
I don't understand the test files. What exactly needs to be done. Do I make changes to the vertices and the triangles functions?

**Comment by skoudoro** [comment]
The tests are making sure that your function returns the expected values. So you have to call your new function and compare the results with what you expect. `assert_*` make this comparison. In your specific case, When we do:
```python
vertices, triangles = prim_tetrahedron()
npt.assert_equal(triangles.shape, (4, 3))
```
We check if `prim_tetahedron` returns the correct number of triangles. If someone else modifies your code and add one more triangle. the test will fail and we will know that something might be wrong on his change. There are many other points to check so I let look at `test_primitives.py` file.
I hope it helps @kakashihatakae.

File: fury/primitive.py [hunk 12]
```diff
@@ -328,3 +331,77 @@ def _fexp(x, p):
...
+def prim_tetrahedron():
+    """
+    returns vertices and triangles to a tetrahedron
+    that has a side length of two units.
+
+
+    Returns
+    -------
+    pyramid_vert: numpy.ndarray array
```
Comment (skoudoro) [inline fury/primitive.py]: twice `array`, can you remove one

File: fury/primitive.py [hunk 13]
```diff
@@ -328,3 +331,77 @@ def _fexp(x, p):
...
+        4 vertices coordinates
+    triangles: numpy.ndarray
+        4 triangles representing the tetrahedron
+    """
+    pyramid_vert = np.array([[0.0, 0.0, 0.0], [2.0, 0.0, 0.0], [1.0, math.sqrt(
```
Comment (skoudoro) [inline fury/primitive.py]: Can you avoid to put everything in one line? Try to follow the same format as `pyramid_triag`

File: fury/primitive.py [hunk 14]
```diff
@@ -328,3 +331,77 @@ def _fexp(x, p):
...
+    return pyramid_vert, pyramid_triag
+
+
+def prim_icosahedron():
+    """
+    returns vertices and triangles to a icosahedron
+
+    Returns
+    -------
+    icosahedron_vertices: numpy.ndarray array
```
Comment (skoudoro) [inline fury/primitive.py]: twice `array`, can you remove one

**Comment by kakashihatakae** [comment]
Changed the vertices of the tetrahedron function, edited comments, and added tests in test_vertices_primitives() and test_triangles_primitives().

File: fury/tests/test_primitive.py [hunk 15]
```diff
@@ -1,24 +1,38 @@
...
 def test_vertices_primitives():
     l_primitives = [(fp.prim_square, (4, 3)),
-                    (fp.prim_box, (8, 3))]
+                    (fp.prim_box, (8, 3)),
+                    (fp.prim_tetrahedron, (4, 3))]
 
     for func, shape in l_primitives:
         vertices, _ = func()
-
+        print('ddfsdf')
```
Comment (skoudoro) [inline fury/tests/test_primitive.py]: Can you remove this `print`

**Comment by skoudoro** [comment]
thank you @kakashihatakae! merging

**Pull request merged**